mantidproject/mantid
# Modify camera toolbar to snap to nonorthogonal axes.

Description of work.

This modifies the ParaView camera toolbar to align the view along the crystallographic instead of cartesian axes. This required reworking the `pqCameraToolbar` and `pqCameraReaction`classes and adding them to the VSI. 

I'm allowing VTK to orthogonalize the up direction because transforming it doesn't seem to produce a better result.

The icons still show X, Y, Z. We may want to change these before the release.

**To test:**



1. Open a MDHistoWorkspace with nonorthogonal axes in the VSI. I used a hexagonal cell (symm.nxs from @AndreiSavici), so another file with different symmetry would be ideal.
2. Click the icons in the camera toolbar and verify that the views are now along the crystallographic axes defined by the change-of-basis matrix.

There is no GitHub issue associated with this pull request.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
